src/main/java/seedu/address/model/person/Person.java

```diff
@@ -31,7 +31,7 @@
     private ObjectProperty<UniqueSocialInfoList> socialInfos;
 
     /**
-     * Every field must be present and not null.
+     * Every field must be present but can be null except 'name'.
      */
```

Comment:
As mentioned above, might be worth handling the null logic here (i.e. this.phone.set(null) or something like that) rather than having exception logic in each of the component classes